src/overview/actions.js

```diff
@@ -31,8 +31,25 @@ export function init() {
 // Search for docs matching the current query, update the results
 export function refreshSearch({loadingIndicator = false}) {
     return async function (dispatch, getState) {
-        const { query, startDate, endDate } = ourState(getState())
-        const oldResult = ourState(getState()).searchResult
+        let {
+            query,
+            startDate: oldStartDate,
+            endDate: oldEndDate,
+            searchResult: oldResult,
+        } = ourState(getState())
+
+        const {
+            startDate: newStartDate,
+            endDate: newEndDate,
+            extractedQuery,
+        } = extractTimeFiltersFromQuery(query)
+
+        const startDate = newStartDate || oldStartDate
+        const endDate = newEndDate || oldEndDate
+        query = extractedQuery || query
+
+        if (oldStartDate !== newStartDate) dispatch(setStartDate({startDate}))
+        if (oldEndDate !== newEndDate) dispatch(setEndDate({endDate}))
```

Comment:
Example makes total sense. This should definitely be changed. There's also a problem for what we should do when a user changes "before" and "after" time filters by using the time pickers. Don't know if we should change the query or what to do... If we don't sync the change to the query the next character that is typed would override date pickers.